impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -121,45 +111,47 @@ private Server buildKafkaServer(KafkaClusterConfig.ConfigHolder c, List<String>
             return instantiateKraftServer(config, threadNamePrefix);
         }
         else {
-            createScramUsersInZookeeper(config, scramArguments);
-            return new KafkaServer(config, Time.SYSTEM, threadNamePrefix, false);
+            var kafkaServerClazz = getKafkaServerClazz();
+            createScramUsersInZookeeper(kafkaServerClazz, scramArguments, config);
+            return instantiateKafkaZookeeperServer(kafkaServerClazz, config, threadNamePrefix);
         }
     }
 
+    @NonNull
+    @SuppressWarnings("unchecked")
+    private Class<Server> getKafkaServerClazz() {
+        try {
+            return (Class<Server>) Class.forName("kafka.server.KafkaServer");
+        }
+        catch (ClassNotFoundException e) {
+            throw new IllegalStateException("Failed to find Kafka Zookeeper Server on classpath", e);
+        }
+    }
+
+    /**
+     * We instantiate the KafkaServer reflectively in order to continue to support older
+     * versions of Kafka that used Zookeeper based controllers.
+     *
+     * @param kafkaServerClazz kafka server class
+     * @param config           kafka config
+     * @param threadNamePrefix thread name prefix to be used by the server
+     */
+    @NonNull
+    private Server instantiateKafkaZookeeperServer(Class<Server> kafkaServerClazz, KafkaConfig config, Option<String> threadNamePrefix) {
```

Comment:
Scala's option was a bit of surprise here. I see that its needed because it gets passed through as is to the construct call.